apps/meteor/server/models/raw/ModerationReports.ts

```diff
@@ -132,12 +135,97 @@ export class ModerationReportsRaw extends BaseRaw<IModerationReport> implements
 		return this.col.aggregate(params, { allowDiskUse: true });
 	}
 
+	findUserReports(
+		latest: Date,
+		oldest: Date,
+		selector: string,
+		pagination: PaginationParams<IModerationReport>,
+	): AggregationCursor<Pick<UserReport, '_id' | 'reportedUser' | 'ts'> & { count: number }> {
+		const query = {
+			_hidden: {
+				$ne: true,
+			},
+			ts: {
+				$lt: latest,
+				$gt: oldest,
+			},
+			message: {
+				$exists: false,
+			},
+			...this.getSearchQueryForSelectorUsers(selector),
+		};
+
+		const { sort, offset, count } = pagination;
+
+		const pipeline = [
+			{ $match: query },
+			{
+				$sort: {
+					ts: -1,
+				},
+			},
+			{
+				$group: {
+					_id: '$reportedUser._id',
+					count: { $sum: 1 },
+					reports: { $first: '$$ROOT' },
+				},
+			},
+			{
+				$sort: sort || {
+					'reports.ts': -1,
+				},
+			},
+			{
+				$skip: offset,
+			},
+			{
+				$limit: count,
+			},
+			{
+				$project: {
+					_id: 0,
+					reportedUser: '$reports.reportedUser',
+					ts: '$reports.ts',
+					count: 1,
+				},
+			},
+		];
+
+		return this.col.aggregate(pipeline, { allowDiskUse: true, readPreference: readSecondaryPreferred() });
+	}
+
 	countMessageReportsInRange(latest: Date, oldest: Date, selector: string): Promise<number> {
-		return this.col.countDocuments({
-			_hidden: { $ne: true },
-			ts: { $lt: latest, $gt: oldest },
-			...this.getSearchQueryForSelector(selector),
-		});
+		return this.col
+			.distinct('message.u._id', {
```

Comment:
this means "unique reports for a message sender" - so 2 different messages sent by the same user would count as one. should this not be all reports?

Reply:
Please rename the model function then. The method there is not saying this :)

Reply:
Thank you, done!